models/classes/Lists/DataAccess/Repository/DependsOnPropertyRepository.php

```diff
@@ -34,6 +42,8 @@
 
 class DependsOnPropertyRepository extends ConfigurableService
 {
+    private const CACHE_PREFIX = 'dependonProperty:';
```

Comment:
You can use a cache mask instead of concatenating all over the places.

```suggestion
    private const CACHE_MASK = 'depends_on_property:%s_%s';
```

And then to retrieve: 

```
$this->getCache()->get(sprintf(self::CACHE_MASK, $property->getUri(), $listUri));
```